### PR TITLE
Ensure skip cert check when sending out raw requests if related env var is configured

### LINKED ISSUE #5099: Not able to use Azure CLI with self-signed certificate
Environment summary

CLI version azure-cli (2.0.22)
OS Type: Windows 10
Installation via: apt-get for Bash on Ubuntu on Windows

I am trying to run cli commands with a self-signed certificate and I am getting following error:

"Please ensure you have network connection. Error detail: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /common/oauth2/devicecode?api-version=1.0 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))"

### LINKED ISSUE #5258: Run Azure CLI with self-signed cert
Environment summary

CLI version azure-cli (2.0.22)
OS Type: Windows 10
Installation via: apt-get for Bash on Ubuntu on Windows

I am trying to create VM using the following command:
az vm create --resource-group anshitagroup --name myVM --location westeurope --nics myNic --image win2016datacenter --adm
in-username ramarao@revskope.onmicrosoft.com --admin-password password

And I am getting the following error:
usage: az vm create [-h] [--output {json,tsv,table,jsonc}] [--verbose]
                    [--debug] [--query JMESPATH]
                    [--data-disk-caching {None,ReadOnly,ReadWrite}]
                    [--assign-identity [ASSIGN_IDENTITY [ASSIGN_IDENTITY ...]]]
                    [--public-ip-address-dns-name PUBLIC_IP_ADDRESS_DNS_NAME]
                    [--scope IDENTITY_SCOPE] [--image IMAGE] [--no-wait]
                    [--storage-sku {Premium_LRS,Standard_GRS,Standard_LRS,Standard_RAGRS,Standard_ZRS}]
                    [--nsg NSG] [--os-disk-name OS_DISK_NAME]
                    [--storage-container-name STORAGE_CONTAINER_NAME]
                    [--validate] [--size SIZE] [--subnet SUBNET]
                    [--availability-set AVAILABILITY_SET]
                    [--nsg-rule {RDP,SSH}] [--zone {1,2,3}]
                    [--attach-data-disks ATTACH_DATA_DISKS [ATTACH_DATA_DISKS ...]]
                    [--nics NICS [NICS ...]]
                    [--authentication-type {ssh,password}]
                    [--admin-password ADMIN_PASSWORD] [--plan-name PLAN_NAME]
                    [--attach-os-disk ATTACH_OS_DISK] [--location LOCATION]
                    --name NAME [--public-ip-address PUBLIC_IP_ADDRESS]
                    [--vnet-name VNET_NAME] [--tags [TAGS [TAGS ...]]]
                    [--ssh-key-value SSH_KEY_VALUE]
                    [--data-disk-sizes-gb DATA_DISK_SIZES_GB [DATA_DISK_SIZES_GB ...]]
                    [--license-type {Windows_Server,Windows_Client}]
                    [--storage-caching {ReadOnly,ReadWrite}]
                    [--storage-account STORAGE_ACCOUNT]
                    [--asgs APPLICATION_SECURITY_GROUPS [APPLICATION_SECURITY_GROUPS ...]]
                    [--use-unmanaged-disk] [--generate-ssh-keys]
                    [--os-disk-size-gb OS_DISK_SIZE_GB]
                    [--plan-product PLAN_PRODUCT]
                    [--subnet-address-prefix SUBNET_ADDRESS_PREFIX]
                    --resource-group RESOURCE_GROUP_NAME
                    [--public-ip-address-allocation {dynamic,static}]
                    [--secrets SECRETS [SECRETS ...]]
                    [--plan-publisher PLAN_PUBLISHER] [--role IDENTITY_ROLE]
                    [--admin-username ADMIN_USERNAME]
                    [--ssh-dest-key-path SSH_DEST_KEY_PATH]
                    [--plan-promotion-code PLAN_PROMOTION_CODE]
                    [--vnet-address-prefix VNET_ADDRESS_PREFIX]
                    [--custom-data CUSTOM_DATA]
                    [--private-ip-address PRIVATE_IP_ADDRESS]
                    [--os-type {windows,linux}]
az vm create: error: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)>

Azure Version details:
azure-cli (2.0.23)

acr (2.0.17)
acs (2.0.22)
advisor (0.1.0)
appservice (0.1.22)
backup (1.0.3)
batch (3.1.7)
batchai (0.1.3)
billing (0.1.6)
cdn (0.0.10)
cloud (2.0.10)
cognitiveservices (0.1.9)
command-modules-nspkg (2.0.1)
configure (2.0.12)
consumption (0.2.0)
container (0.1.15)
core (2.0.23)
cosmosdb (0.1.15)
dla (0.0.15)
dls (0.0.18)
eventgrid (0.1.5)
extension (0.0.6)
feedback (2.0.6)
find (0.2.7)
interactive (0.3.11)
iot (0.1.15)
keyvault (2.0.15)
lab (0.0.13)
monitor (0.0.13)
network (2.0.19)
nspkg (3.0.1)
profile (2.0.16)
rdbms (0.0.9)
redis (0.2.10)
reservations (0.1.0)
resource (2.0.19)
role (2.0.15)
servicefabric (0.0.7)
sql (2.0.17)
storage (2.0.21)
vm (2.0.20)

Python location '/usr/bin/python'
Extensions directory '/home/anshita/.azure/cliextensions'

Python (Linux) 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609]

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #5099. Fix #5258 
@derekbekoe, I can also update related [extension code](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/_index.py#L26). So let me know 
@djyou,  I can update [acr code](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py#L57). So let me know

Moving forward, during code review, for any HTTTPS raw requests outside of Python SDK, we just ensure the new code checks the global flag

- [will do later] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [na] Each command and parameter has a meaningful description.
- [na] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by djyou** [comment]
Thank you @yugangw-msft Please go ahead with updating acr code.

**Comment by djyou** [comment]
`acr` part looks good to me.

**Comment by yugangw-msft** [comment]
@derekbekoe  @djyou, thanks for the review. I have tried `mitmproxy` on ubuntu using mitmproxy's default unsigned cert, it works now to capture traces from commands like `az extension list-available` and `az acr repository list -n ygacr`. Of course, `az vm create` also works.
Merging.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py [hunk 1]
```diff
@@ -72,14 +71,19 @@ def _load_images_from_publisher(publisher):
...
+    from azure.cli.core.util import should_disable_connection_verify
     try:
         target_url = cli_ctx.cloud.endpoints.vm_image_alias_doc
     except CloudEndpointNotSetException:
         raise CLIError("'endpoint_vm_image_alias_doc' isn't configured. Please invoke 'az cloud update' to configure "
                        "it or use '--all' to retrieve images from server")
-    txt = urlopen(target_url).read()
-    dic = json.loads(txt.decode())
+    # under hack mode(say through proxies with unsigned cert), opt out the cert verification
+    response = requests.get(target_url, verify=(not should_disable_connection_verify()))
```
Comment (zhoxing-ms) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py:83]: This is where ` urlopen` was modified

**Pull request merged**